reactor-core/src/test/java/reactor/core/scheduler/ParallelSchedulerTest.java

```diff
@@ -40,16 +43,36 @@ protected Scheduler scheduler() {
 		return Schedulers.newParallel("ParallelSchedulerTest");
 	}
 
-	@Test(expected = IllegalArgumentException.class)
-	public void negativeParallelism() throws Exception {
-		Schedulers.newParallel("test", -1);
-	}
-
 	@Override
 	protected boolean shouldCheckInterrupted() {
 		return true;
 	}
 
+	@Test
+	public void startAndDecorationImplicit() {
+		AtomicInteger decorationCount = new AtomicInteger();
+		Schedulers.setExecutorServiceDecorator("startAndDecorationImplicit", (s, srv) -> {
+			decorationCount.incrementAndGet();
+			return srv;
+		});
+		final Scheduler scheduler = afterTest.autoDispose(new ParallelScheduler(4, Thread::new));
+		afterTest.autoDispose(() -> Schedulers.removeExecutorServiceDecorator("startAndDecorationImplicit"));
+
+		assertThat(decorationCount).as("before schedule").hasValue(0);
+		//first scheduled task implicitly starts the scheduler and thus creates _parallelism_ workers/executorServices
+		scheduler.schedule(ThrowingRunnable.unchecked(() -> Thread.sleep(100)));
+		assertThat(decorationCount).as("after schedule").hasValue(4);
```

Comment:
nit: it'd be better to extract the magical `4` into a variable (something like `parallelism`) to make it clear that we're referring to the same value as constructor's parameter

Reply:
done